Public/Src/Cache/ContentStore/App/Hello.cs

```diff
@@ -32,18 +32,25 @@ internal sealed partial class Application
             [Description("GRPC port on the target machine"), DefaultValue(ServiceConfiguration.GrpcDisabledPort)] int grpcPort,
             [Description("Name of the memory mapped file used to share GRPC port. 'CASaaS GRPC port' if not specified.")] string grpcPortFileName)
         {
-            Initialize();
-
-            if (grpcPort == 0)
+            try
             {
-                grpcPort = Helpers.GetGrpcPortFromFile(_logger, grpcPortFileName);
-            }
+                Initialize();
+
+                if (grpcPort == 0)
+                {
+                    grpcPort = Helpers.GetGrpcPortFromFile(_logger, grpcPortFileName);
+                }
 
-            var _channel = new Channel(host, grpcPort, ChannelCredentials.Insecure);
-            var _client = new ContentServer.ContentServerClient(_channel);
-            var helloResponse = _client.Hello(new HelloRequest(), new CallOptions(deadline: DateTime.UtcNow + TimeSpan.FromSeconds(2)));
+                var _channel = new Channel(host, grpcPort, ChannelCredentials.Insecure);
+                var _client = new ContentServer.ContentServerClient(_channel);
+                var helloResponse = _client.Hello(new HelloRequest(), new CallOptions(deadline: DateTime.UtcNow + TimeSpan.FromSeconds(2)));
 
-            _logger.Always($"Hello response {(helloResponse.Success ? "succeeded" : "failed")}: {helloResponse.ToString()}");
+                _logger.Always("Hello response {0}: {1}", helloResponse.Success ? "succeeded" : "failed", helloResponse.ToString());
+            }
+            catch (Exception e)
+            {
+                _logger.Always(e.ToString());

```

Comment:
I can't remember the semantics of this method: will it be visible to the user in interactive mode?

Reply:
I think we should definitely change the output message here to something: "Unhandled error occurred in Hello" something... I've seen some cases in Kusto when it is not clear what was the error. And I think it make sense to print only message to the console and log the entire exception.

---
In reply to: [277806629](https://github.com/Microsoft/BuildXL/pull/185#discussion_r277806629) [](ancestors = 277806629)

Reply:
This error will never surface in Kusto, because it will never be run in any official capacity. This is a toy verb like HardlinkBenchmark and CopyFile.

---
In reply to: [277811685](https://github.com/Microsoft/BuildXL/pull/185#discussion_r277811685) [](ancestors = 277811685,277806629)